BUILDING.md

```diff
@@ -85,3 +85,23 @@ Example:
 
 If your port/build includes `arm-none-eabi-gdb-py`, consider using it instead, as it can be used for better register
 debugging with https://github.com/bnahill/PyCortexMDebug
+
+# Code Quality Checks
+
+We apply code quality checks using pre-commit.  Install pre-commit once per system with
+
+    python3 -mpip install pre-commit
+
+Activate it once per git clone with
+
+    pre-commit --install
+
+Pre-commit also requires some additional programs to be installed through your package manager:
+
+ * Standard Unix tools such as make, find, etc
+ * The gettext package, any modern version
+ * uncrustify version 0.71 (0.72 is also tested)
```

Comment:
I think we should point users to only `0.71.0`.

Reply:
Arch and Debian only provide easy access to 0.72.  We want to support Arch and Debian users, if it is feasible, rather than forcing use of Ubuntu.

Reply:
I think the uncrustify configuration while is version specific.
I fear running different versions might introduce conflicting changes.
Are you running `0.72`?

Reply:
Yes, both @tannewt and I used 0.72 locally.  I used 0.72 locally when preparing this PR, and upon upload the CI ran 0.71 and didn't detect introduce any formatting changes during pre-commit.

Reply:
Since I saw pre-commit with 0.72 introduce some changes compared to what you had committed (d3bf1fe15dd3720d6566e061a0d315add550b248) I verified that the same changes are introduced by 0.71.  I do not know why these changes were not previously introduced, except possibly due to problems running on the right sets of files before the changes in this PR.

Reply:
Okay, thanks for clarifying that. :)